### ISSUE TITLE
Data caching issue for writing and reading files in Windows container

### ISSUE THREAD

**Issue by jingxu97** [kubernetes/kubernetes#81690]
While investigating one flaky test for in-tree GCE PD in Windows "should store data"
https://testgrid.k8s.io/sig-windows#gce-windows-master&width=20&include-filter-by-regex=should%20store%20data, we noticed the following behavior.

1. Start a pod with a GCE PD volume, and write some content to a file (use echo command), and verify the content by reading the file
2. Kill the pod
3. Start a new Pod which uses the same volume and read the content of the file. 

The test sometimes fails, and it seems caused by data caching. When the first pod is killed, the content has not yet flushed to the disk when the disk is detached, so that when the second pod trying to read the file, the content is empty. 

The relevant test code 
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/testsuites/volumes.go#L145
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/volume/fixtures.go#L638

**Comment by yujuhong** [comment]
/sig windows

Looks like we don't run [mountvol](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/mountvol) at all in kubernetes. But I guess this should be called when detaching the PD?

@pjh do you know what commands are run when detaching a PD?

**Comment by pjh** [comment]
> @pjh do you know what commands are run when detaching a PD?

I'm not sure. I did a brief search in our GCE documentation and didn't find any recommendations about detaching PDs, only about [attaching](https://cloud.google.com/compute/docs/disks/add-persistent-disk#create_disk) them.

I took a look at some GCE test code that we have and it looks like Windows (at least older versions) doesn't have a native sync command, but there is a sync.exe tool in the well-known [SysInternals](https://docs.microsoft.com/en-us/sysinternals/downloads/sync) package. If we're desperate we could try putting that in the test container and running it before detach but I don't know if it would do anything.
